### ISSUE TITLE
Convention for the complex derivative

### ISSUE THREAD

**Issue by Jakob-Unfried** [google/jax#1853]
I know of two natural ways to talk about complex derivatives:
1. The [Wirtinger derivative](https://en.wikipedia.org/wiki/Wirtinger_derivatives)
For a complex number z = x + iy and a function f(z) we have two derivatives
∂f/∂z = (∂f/∂x - i ∂f/∂y)/2  and ∂f/∂z* = (∂f/∂x + i ∂f/∂y)/2

2. The "gradient"
∇_z(f) = ∂f/∂x + i ∂f/∂y = 2 ∂f/∂z*
which is useful in gradient descend algorithms, since it gives consists of the components of the gradient, when interpreting f as a function of two real variables f(x, y) and thus gives the direction of the steepest descend when f is a real function.

I noticed that `jax.grad` follows neither of these and instead computes ∂f/∂x - i ∂f/∂y = 2∂f/∂z
This is readily checked since `grad(np.abs)(1j) == -1j`

I feel like this is counter-intuitive and want to start a discussion, if this is indeed the behaviour that Jax should have.

I also suggest incorporating this in the Gotcha's page

**Comment by hawkinsp** [comment]
@dougalm is the expert here, but please take a look at
https://colab.research.google.com/github/google/jax/blob/master/docs/notebooks/autodiff_cookbook.ipynb#scrollTo=3fPWLrxK8bgD

**Comment by Jakob-Unfried** [comment]
@hawkinsp 
Thanks for the pointer and sorry for letting you google that for me.

I am reasonably convinced by the given arguments, that this is a good behaviour, and most importantly it is well-explained in that notebook.

So the remainder of this issue is my suggestion to add a note in the Gotcha's to point to this part of the Autodiff Cookbook

**Comment by hawkinsp** [comment]
Can you suggest a good place in the documentation you would have seen this?

My guess is that the documentation of `jax.grad`
https://jax.readthedocs.io/en/latest/jax.html#automatic-differentiation
would be a good place to describe or link to a description of the convention for complex numbers. Would that have worked?

It's not clear to me the `gotchas` list on the GitHub readme is the right place.

**Comment by Jakob-Unfried** [comment]
I would add a short description and a link to the cookbook to the documentation of `jax.grad`

I suggest appending something like this at the end of the **Returns:** block, i.e. a shorter version of what is written in the cookbook

> If `fun` is a real function of complex inputs then the conjugate of `grad(fun)(x)` is the direction of steepest ascend. If `fun` is a complex holomorphic function the output agrees with the usual holomorphic derivative. For more details see [The Autodiff Cookbook](https://colab.research.google.com/github/google/jax/blob/master/docs/notebooks/autodiff_cookbook.ipynb#scrollTo=3fPWLrxK8bgD)

While i would personally have expected a similar pointer/link in [the common gotchas list](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html),
i don't see that as strictly necessary.
Maybe the behaviour is less of a surprise for other people and i just had a rigid opinion on what to expect from `grad` of a complex function
